PocketKit/Sources/PocketKit/Home/HomeViewModel.swift

```diff
@@ -440,111 +257,255 @@ extension HomeViewModel {
 
             tracker.track(
                 event: ContentOpenEvent(destination: .internal, trigger: .click),
-                contexts(for: cell, at: indexPath)
+                contexts(for: savedItem, at: indexPath)
             )
         }
     }
+}
 
-    private func report(_ cell: HomeViewModel.Cell, at indexPath: IndexPath) {
-        guard let objectID = getRecommendationID(with: cell),
-              let viewModel = viewModel(for: objectID) else {
-            return
+// MARK: - Section Headers
+extension HomeViewModel {
+    func sectionHeaderViewModel(for section: Section) -> SectionHeaderView.Model? {
+        switch section {
+        case .recentSaves:
+            return .init(name: "Recent Saves", buttonTitle: "My List") { [weak self] in
+                self?.tappedSeeAll = .myList
+            }
+        case .slateHero(let objectID):
+            guard let slate = source.mainContext.object(with: objectID) as? Slate else {
+                return nil
+            }
+
+            return .init(name: slate.name ?? "", buttonTitle: "See All") { [weak self] in
+                self?.select(slate: slate)
+            }
+        case .loading, .slateCarousel:
+            return nil
+        }
+    }
+}
+
+// MARK: - Recent Saves Model & Actions
+extension HomeViewModel {
+    func numberOfRecentSavesItem() -> Int {
+        return 5
```

Comment:
do we want to use the `recentSaves.count` from the fetchRequest or do we know that our limit is 5, so it's okay to put 5 here? Maybe add a constant to be used in both places?

Reply:
Good catch, we should be returning the actual number of recents saves in the database